### PR TITLE
Expose SFrame errors through SFrameTransform.onerror

### PR DESCRIPTION
**Pull request by youennf**
***
<a href="https://pr-preview.s3.amazonaws.com/youennf/webrtc-insertable-streams/pull/103.html" title="Last updated on Jun 4, 2021, 9:33 AM UTC (3da3736)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-encoded-transform/103/1922fed...youennf:3da3736.html" title="Last updated on Jun 4, 2021, 9:33 AM UTC (3da3736)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -217,18 +239,29 @@ The <dfn constructor for="SFrameTransform" lt="SFrameTransform(options)"><code>n
...
+1. If |frame|.`[[owner]]` is a {{RTCRtpSender}}, set |role| to 'encrypt'.
+1. If |frame|.`[[owner]]` is a {{RTCRtpReceiver}}, set |role| to 'decrypt'.
+1. Let |data| be undefined.
+1. If |frame| is a {{BufferSource}}, set |data| to |frame|.
+1. If |frame| is a {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
+1. If |frame| is a {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
+1. If |data| is undefined, abort these steps.
+1. Let |buffer| be the result of running the SFrame algorithm with |data| and |role| as parameters. This algorithm is defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a> and returns an {{ArrayBuffer}}.
+1. If the SFrame algorithm exit abruptly with an error, run the following sub steps:
+     1. If the processing fails on decryption side due to |data| not following the SFrame format, [=fire an event=] named {{SFrameTransform/onerror|error}} at |sframe|,
```
Comment (alvestrand) [inline index.bs:251]: "Fire an event" is synchronous. Is this intended to block processing?

Comment (youennf) [reply index.bs:251]: Right, forgot queue a task, thanks!

**Comment by youennf** [comment]
Fixes https://github.com/w3c/webrtc-encoded-transform/issues/58

File: index.bs [hunk 2]
```diff
@@ -217,18 +239,29 @@ The <dfn constructor for="SFrameTransform" lt="SFrameTransform(options)"><code>n
...
+1. If |frame| is a {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
+1. If |frame| is a {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
+1. If |data| is undefined, abort these steps.
+1. Let |buffer| be the result of running the SFrame algorithm with |data| and |role| as parameters. This algorithm is defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a> and returns an {{ArrayBuffer}}.
+1. If the SFrame algorithm exit abruptly with an error, [=queue a task=] to run the following sub steps:
+     1. If the processing fails on decryption side due to |data| not following the SFrame format, [=fire an event=] named {{SFrameTransform/onerror|error}} at |sframe|,
+        using the {{SFrameTransformErrorEvent}} interface with its {{SFrameTransformErrorEvent/type}} attribute set to {{SFrameTransformErrorEventType/syntax}}
+        and its {{SFrameTransformErrorEvent/frame}} attribute set to |frame|.
+     1. If the processing fails on decrpytion side due to the key identifier parsed in |data| being unknown, [=fire an event=] named {{SFrameTransform/onerror|error}} at |sframe|,
```
Comment (jan-ivar) [inline index.bs:254]:
```suggestion
     1. If the processing fails on decryption side due to the key identifier parsed in |data| being unknown, [=fire an event=] named {{SFrameTransform/onerror|error}} at |sframe|,
```

**Review by jan-ivar — APPROVED** [review]

**Comment by youennf** [comment]
MediaStreamTrack link issue is pre-existing to this PR. Merging now.
@dontcallmedom, do you know what the issue is?

**Comment by dontcallmedom** [comment]
I suspect this is an indirect consequence of https://github.com/w3c/respec/issues/3592 which will likely take some time to settle downstream :(

**Pull request merged**